.github/workflows/tests-ipsec-upgrade.yaml

```diff
@@ -322,16 +321,11 @@ jobs:
               --junit-file "cilium-junits/${{ env.job_name }} (${{ join(matrix.*, ', ') }}).xml" \
               --junit-property github_job_step="Run tests upgrade 1 (${{ join(matrix.*, ', ') }})"
 
-            # Create pods which establish long lived connections. It will be used by
-            # subsequent connectivity tests with --include-conn-disrupt-test to catch any
-            # interruption in such flows.
-            ./cilium-cli connectivity test --include-conn-disrupt-test --conn-disrupt-test-setup
-
-      - name: Upgrade Cilium (${{ matrix.name }})
-        uses: cilium/little-vm-helper@6a3d1797af8fab1a49f1c3d09afe3230aea559b6 # v0.0.11
+      - name: Upgrade Cilium & Test (${{ matrix.name }})
+        uses: ./.github/actions/conn-disrupt-test
```

Comment:
I'm under the impression this action gets executed from the PR branch. If so, as a malicious actor I can put stuff in the commands of the action and get code execution in a PR, no?

Reply:
I'm not sure but I expect only a [member of cilium](https://github.com/cilium/community/blob/main/ladder/members.yaml) has permission to create a branch?

Reply:
I believe David is correct. The only other time we execute a checked out script (`./contrib/scripts/kind.sh`) in this workflow is within a VM (LVH). Might be worth getting a confirmation from @nbusseneau.

If this is insecure, then I guess our (only?) option is to move this new GitHub action to its own repository. Or can we use a specific version (specific SHA) of this new GitHub action without moving it elsewhere?

Reply:
That is correct, we must be careful with how local `uses` are used in privileged contexts such as `workflow_dispatch`-based workflows, since they have access to repository secrets and also have extended permissions.

If you grep for `uses: .` you can inspect how do we do it for the existing composite actions (for env variables and EKS configuration): we make sure the local `uses` are only done in the "trusted" part of the workflow, and we do not use them in the "untrusted" part (which is everything after we checked out the PR code, see line 234 in this file for example).

I was thinking the other day that we should remove local `uses` entirely and switch to a static reference anyway, that would help with not having to concern ourselves with this specific issue for maintenance and reviews, so I would advocate for doing so here:

```suggestion
        uses: ${{ github.repository }}/.github/actions/conn-disrupt-test@${{ inputs.SHA || github.sha }}
```

This should pull the composite actions from the appropriate context based on trigger (if `workflow_dispatch`, `input.SHA` will be either the PR target branch or the PR branch itself depending on whether the PR comes from a fork or not, and if `schedule`, `github.sha` will be used as fallback).

Reply:
Unfortunately `uses: ${{ github.repository }}/.github/actions/conn-disrupt-test@${{ inputs.context-ref || github.sha }}` was invalid: https://github.com/cilium/cilium/actions/runs/5889885321
```

Invalid workflow file: .github/workflows/tests-ipsec-upgrade.yaml#L333
The workflow is not valid. .github/workflows/tests-ipsec-upgrade.yaml (Line: 333, Col: 15): Unrecognized named-value: 'github'. Located at position 1 within expression: github.repository .github/workflows/tests-ipsec-upgrade.yaml (Line: 352, Col: 15): Unrecognized named-value: 'github'. Located at position 1 within expression: github.repository
```

Then I decide to create a independent PR https://github.com/cilium/cilium/pull/27567 to add this new action, so we can refer to the action using `cilium/cilium/.github/actions/conn-disrupt-test@SHA` to address the security concern.